header-rewrite-filter/header_processor.cc

```diff
@@ -51,36 +60,40 @@ namespace HeaderRewriteFilter {
         // call ConditionProcessor executeOperation; if it is null, return true
         bool result = true;
         if (getConditionProcessor()) {
-            result = getConditionProcessor()->executeOperation();
+            const absl::Status status = getConditionProcessor()->executeOperation(getConditionProcessor()->getBoolProcessors());
+            if (status != absl::OkStatus()) {
+                return status;
+            }
+            result = getConditionProcessor()->getConditionResult();
         }
         setCondition(result);
 
         return absl::OkStatus();
     }
 
-    absl::Status HeaderProcessor::ConditionProcessorSetup(std::vector<absl::string_view>& condition_expression, std::vector<absl::string_view>::iterator start) {
-        if (start == condition_expression.end()) {
-            return absl::InvalidArgumentError("empty condition provided");
+    absl::Status SetHeaderProcessor::executeOperation(Http::RequestOrResponseHeaderMap& headers, SetBoolProcessorMapSharedPtr bool_processors) {
+        const ConditionProcessorSharedPtr condition_processor = getConditionProcessor();
+        if (condition_processor) {
```

Comment:
I think it could be simpler to pass in a bool processor that always evals to true if there's no condition processor. Right now you make a decision in `SetHeader::executeOperation` and also in `HeaderProcessor::evaluateCondition`